.github/workflows/tests.yml

```diff
@@ -1,31 +1,35 @@
-name: Test
+name: Tests
 
 on:
   push:
     branches: [main]
   pull_request:
-    # Check all PR
+    branches: [main]
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
+  tests:
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10"]
+        os: ['ubuntu-latest']
+        python-version: ['3.8', '3.9', '3.10', '3.11']
+        include:
+          - os: macos-latest
+            python-version: '3.10'
+          - os: windows-latest
+            python-version: '3.10'
```

Comment:
Given that this is a pure Python package, the gain of checking code on different OS is much less than for packages with native code. I'm no against adding more builders, as long as they are fast and don't block CI

Reply:
works for me